benchmarks/lbfgs_benchmark.py

```diff
@@ -108,6 +114,91 @@ def run_multiclass_logreg():
   plt.show()
 
 
+def cnn(linesearch):
+
+  def load_dataset(dataset, batch_size):
+    """Loads the dataset as a generator of batches."""
+    train_ds, ds_info = tfds.load(f"{dataset}:3.*.*", split="train",
+                                  as_supervised=True, with_info=True)
+    train_ds = train_ds.repeat()
+    train_ds = train_ds.shuffle(10 * batch_size, seed=0)
+    train_ds = train_ds.batch(batch_size)
+    return tfds.as_numpy(train_ds), ds_info
+
+  class CNN(nn.Module):
+    """A simple CNN model."""
+    num_classes: int
+    net_width: int
+
+    @nn.compact
+    def __call__(self, x):
+      x = nn.Conv(features=self.net_width, kernel_size=(3, 3))(x)
+      x = nn.relu(x)
+      x = nn.avg_pool(x, window_shape=(2, 2), strides=(2, 2))
+      x = nn.Conv(features=self.net_width*2, kernel_size=(3, 3))(x)
+      x = nn.relu(x)
+      x = nn.avg_pool(x, window_shape=(2, 2), strides=(2, 2))
+      x = x.reshape((x.shape[0], -1))  # flatten
+      x = nn.Dense(features=self.net_width*4)(x)
+      x = nn.relu(x)
+      x = nn.Dense(features=self.num_classes)(x)
+      return x
+
+  # Hide any GPUs from TensorFlow. Otherwise TF might reserve memory and make
+  # it unavailable to JAX.
+  tf.config.experimental.set_visible_devices([], 'GPU')
```

Comment:
This is interesting - why do we have TF here at all? Is it because we import tfds above?

Reply:
I believe this is due to tensorflow datasets, which imports (parts of) tensorflow. Without this line, tf occupies the memory of the GPU and jax can't use it